azurerm/internal/services/media/media_private_endpoint_connection.go

```diff
@@ -0,0 +1,190 @@
+//https://docs.microsoft.com/en-us/rest/api/media/privateendpointconnections/createorupdate
+
+package media
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/mediaservices/mgmt/2020-05-01/media"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/media/parse"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceMediaPrivateEndpointConnection() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMediaPrivateEndpointConnectionCreate,
+		Read:   resourceMediaPrivateEndpointConnectionRead,
+		//Update: resourceMediaPrivateEndpointConnectionUpdate,
+		Delete: resourceMediaPrivateEndpointConnectionDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			//Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.PrivateEndpointConnectionID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^[-a-zA-Z0-9(_)]{1,128}$"),
+					"Private Endpoint Connection name must be 1 - 128 characters long, can contain letters, numbers, underscores, and hyphens (but the first and last character must be a letter or number).",
+				),
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"media_services_account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: ValidateMediaServicesAccountName,
+			},
+
+			"private_link_connection_state": {
+				Type:     schema.TypeList,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"description": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"actions_required": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"provisioning_state": {
```

Comment:
Is this attribute needed for any reason? I could understand exposing it if there was `Update` functionality, but if there is no real reason to expose this we can prolly remove this from the schema.

Reply:
> @brunhil, thank you for the PR. It mostly LGTM, but I have left a few minor comments and would like to see some documentation created for this as well before can get this merged.
> 
> Is there a reason why you commented out the `update` function? Does the API not support this functionality?

Thanks for taking a look @WodansSon ! I actually opened and issue on Azure REST API Specs due to missing functionality, https://github.com/Azure/azure-rest-api-specs/issues/12976. It was updated a little while back,

> @brunhil - This is result of private endpoint implementation not being completed yet for Azure Media Services. We are tracking towards having implementation complete in first half of this year.

Reply:
> Thanks for taking a look @WodansSon ! I actually opened and issue on Azure REST API Specs due to missing functionality, [Azure/azure-rest-api-specs#12976](https://github.com/Azure/azure-rest-api-specs/issues/12976). It was updated a little while back,
> 
> > @brunhil - This is result of private endpoint implementation not being completed yet for Azure Media Services. We are tracking towards having implementation complete in first half of this year.

@brunhil, since there seems to be an issue with the REST API and this hasn't been fully baked yet by the service team would it be safe to say we can close this PR as it isn't yet ready to GA?